lib/ed25519.js

```diff
@@ -36,6 +44,140 @@ ed25519.constants.SEED_BYTE_LENGTH = 32;
 ed25519.constants.SIGN_BYTE_LENGTH = 64;
 ed25519.constants.HASH_BYTE_LENGTH = 64;
 
+/**
+ * ASN1 public key validator for importing ed25519 keys in DER format.
+ * @type {Object}
+*/
+var ed25519PublicKeyValidator = {
+  name: 'ed25519PublicKey',
+  tagClass: asn1.Class.UNIVERSAL,
+  type: asn1.Type.SEQUENCE,
+  constructed: true,
+  composed: true,
+  value: [{
+    tagClass: asn1.Class.UNIVERSAL,
+    type: asn1.Type.SEQUENCE,
+    constructed: true,
+    composed: true,
+    value: [{
+      tagClass: asn1.Class.UNIVERSAL,
+      type: asn1.Type.OID,
+      constructed: false,
+      composed: false,
+      capture: 'ed25519Oid'
+    }]
+  }, {
+    tagClass: asn1.Class.UNIVERSAL,
+    type: asn1.Type.BITSTRING,
+    constructed: false,
+    composed: true,
+    captureBitStringValue: 'ed25519PublicKey'
+  }]
+};
+
+/**
+ * ASN1 private key validator for importing ed25519 keys in DER format.
+ * This validator is for a `OneAsymmetricKey` ASN.1 type as specified in
+ * RFC5958 (https://tools.ietf.org/html/rfc5958). This type has the same
+ * format as a simple `PrivateKeyInfo` for which there is already a
+ * validator specified in forge's `rsa` module.
+ *
+ * @type {Object}
+ */
+// TODO: Reuse `PrivateKeyInfo` validator from forge `rsa`
+// https://github.com/digitalbazaar/forge/blob/master/lib/rsa.js#L96
+var ed25519PrivateKeyValidator = {
+  name: 'ed25519PrivateKey',
+  tagClass: asn1.Class.UNIVERSAL,
+  type: asn1.Type.SEQUENCE,
+  constructed: true,
+  composed: true,
+  value: [{
+    tagClass: asn1.Class.UNIVERSAL,
+    type: asn1.Type.INTEGER,
+    constructed: false,
+    composed: false,
+    // only version 1 (which is value `0`) is supported
+    capture: 'privateKeyVersion'
+  }, {
+    tagClass: asn1.Class.UNIVERSAL,
+    type: asn1.Type.SEQUENCE,
+    constructed: true,
+    composed: true,
+    value: [{
+      tagClass: asn1.Class.UNIVERSAL,
+      type: asn1.Type.OID,
+      constructed: false,
+      composed: false,
+      capture: 'ed25519Oid'
+    }]
+  }, {
+    tagClass: asn1.Class.UNIVERSAL,
+    type: asn1.Type.OCTETSTRING,
+    constructed: false,
+    composed: false,
+    capture: 'ed25519PrivateKey'
+  }]
+};
+
+/**
+ * Used to extract the asn1 from DER formatted key.
+ *
+ * @param {NativeBuffer} keyBuffer - A DER formatted key.
+ *
+ * @return {Object} An ASN1 object.
+ */
+function getKeyBytes(keyBuffer) {
+  var binary = forge.util.binary.raw.encode(keyBuffer);
+  return forge.asn1.fromDer(binary);
+}
+
+/**
+ * Takes in a public key exported in DER and returns the raw key value.
+ *
+ * @param {NativeBuffer} publicKeyBuffer - A DER formatted public key.
```

Comment:
Any use of `NativeBuffer` in the docs should change -- this isn't a public name. It should be `{Uint8Array|Buffer}`.

Reply:
This actually is a jsdoc thing NativeBuffer here refers to a jsdoc type declared earlier in the file:

```
/**
 * NativeBuffer is either a Buffer or a Uint8Array.
 * @type {Buffer|Uint8Array}
*/
```